### PR TITLE
Fix all gcc warnings in hrepack

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: mfhdf/hrepack/hrepack.c [hunk 1]
```diff
@@ -342,7 +342,7 @@ int
 read_info(const char *filename, options_t *options)
 {
     char  stype[10];
-    char  comp_info[1024];
+    char  info[1024]; /* compression info */
```
Comment (derobins) [inline mfhdf/hrepack/hrepack.c:345]: comp_info is also a type

File: mfhdf/hrepack/hrepack_check.c [hunk 2]
```diff
@@ -247,13 +247,13 @@ get_scomp(comp_coder_t code)
 static const char *
 get_schunk(int32 flags)
 {
-    if (flags == (HDF_CHUNK | HDF_COMP))
-        return "HDF_CHUNK | HDF_COMP";
-    else if (flags == (HDF_CHUNK))
+    if ((flags & HDF_CHUNK) && (flags & HDF_COMP))
```
Comment (derobins) [inline mfhdf/hrepack/hrepack_check.c:250]: @byrnHDF - I think you meant '&' here?

File: mfhdf/hrepack/hrepack_gr.c [hunk 3]
```diff
@@ -546,10 +550,18 @@ copy_gr(int32 infile_id, int32 outfile_id, int32 gr_in, int32 gr_out, int32 tag,
...
+            printf("Failed to close SDS <%s>\n", path);
+        else
+            printf("Failed to close SDS\n");
+    }
+    if (GRendaccess(ri_out) == FAIL) {
+        if (path)
+            printf("Failed to close SDS <%s>\n", path);
+        else
+            printf("Failed to close SDS\n");
+    }
```
Comment (derobins) [inline mfhdf/hrepack/hrepack_gr.c:564]: Possible use after free of `path`

File: mfhdf/hrepack/hrepack_list.c [hunk 4]
```diff
@@ -507,24 +506,25 @@ list_vg(int32 infile_id, int32 outfile_id, int32 sd_id, int32 sd_out, int32 gr_i
...
 
     if (Vend(infile_id) == FAIL) {
-        printf("Error: Could not end group interface in <%s>\n", vg_name);
+        printf("Error: Could not end infile group interface\n");
         return FAIL;
     }
     if (options->trip == 1) {
         if (Vend(outfile_id) == FAIL) {
-            printf("Error: Could not end group interface in <%s>\n", vg_name);
+            printf("Error: Could not end outfile group interface\n");
```
Comment (derobins) [inline mfhdf/hrepack/hrepack_list.c:520]: Possible use after free of `vg_name`

File: mfhdf/hrepack/hrepack_sds.c [hunk 5]
```diff
@@ -849,7 +849,11 @@ get_print_info(int chunk_flags, HDF_CHUNK_DEF *chunk_def, /* chunk definition */
     if (SDendaccess(sds_id) == FAIL)
         goto out;
 
-    memset(comp_str, 0, 255);
+    /* Has to hold a double, which can have a LOT of digits. Even if that
+     * isn't realistic in practice, the compiler will complain.
+     */
+    if (NULL == (comp_str = calloc(1, 512)))
+        goto out;
```
Comment (derobins) [inline mfhdf/hrepack/hrepack_sds.c:856]: Complaint by the compiler about trying to stuff a double in here

File: mfhdf/hrepack/hrepack_vs.c [hunk 6]
```diff
@@ -40,18 +40,32 @@ copy_vs(int32 infile_id, int32 outfile_id, int32 tag, /* tag of input VS */
...
         options_t *options, list_table_t *list_tbl, int is_lone)
 {
-    int32 vdata_id, /* vdata identifier */
-        vdata_out,  /* vdata identifier */
-        vdata_ref,  /* reference number of the vdata */
-        n_records,  /* number of records */
+    int32 vdata_id  = FAIL; /* vdata identifier */
+    int32 vdata_out = FAIL; /* vdata identifier */
+    int32 vdata_ref,        /* reference number of the vdata */
+        n_records,          /* number of records */
```
Comment (derobins) [inline mfhdf/hrepack/hrepack_vs.c:46]: Large array/stack/frame size warnings

File: mfhdf/hrepack/hrepacktst.c [hunk 7]
```diff
@@ -413,9 +413,11 @@ add_gr(const char  *gr_name,     /* gr name */
...
         dim_gr[2],      /* dimension sizes of the image array */
         interlace_mode, /* interlace mode of the image */
-        data_type,      /* data type of the image data */
-        data[Y_DIM_GR][X_DIM_GR];
-    int           i, j, n = 0, ncomps = 1;
+        data_type;      /* data type of the image data */
+
+    int32        *data   = NULL; /* [Y_DIM_GR][X_DIM_GR] */
+    int           n      = 0;
+    int           ncomps = 1;
```
Comment (derobins) [inline mfhdf/hrepack/hrepacktst.c:420]: More large array/stack/frame size warnings throughout the file

File: mfhdf/hrepack/hrepack_check.c (hunk 2 above)
Comment (byrnHDF) [reply mfhdf/hrepack/hrepack_check.c:250]: probably

**Comment by byrnHDF** [comment]
Why is it that these changes look like ones we already did last time - Just so many of the same type of errors.

**Pull request merged**